### DISCUSSION TITLE
create-external-cluster-resources.py parses only mon leader endpoint

### DISCUSSION THREAD

**Discussion by der-ali** [rook/rook#10605] [Q&A] [answered]
Is there any reason why the `create-external-cluster-resources.py` script parses only the endpoint of the [mon leader ](https://github.com/rook/rook/blob/master/deploy/examples/create-external-cluster-resources.py#L633)? How would the operator watch the mons in the external Ceph, if the leader is down?
@parth-gr

**Comment by parth-gr** [comment, marked as answer]
So once the connection with the rook external cluster is established with ceph cluster,
The rook operator will go and reconcile and will update the mon-endpoint cm with all the mon's that are in the cluster.

There is no specific reason, but it's like a single leader mon is just needed in helping during external-cluster connection so that's why this is only returned.

**Comment by der-ali** [comment]
Ah, i understand. The operator uses the data field in the configmap to establish the connection to the external cluster and then adds the field `csi-cluster-config-json` to the configMap after reconcilation which contains the all the mons.
